q2_mystery_stew/plugin_setup.py

```diff
@@ -129,11 +127,34 @@ def __call__(self, use):
             if argument is None:
                 inputs[name] = transformed_inputs[name] = None
             elif is_semantic_type(template.qiime_type):
-                inputs[name] = use.init_data(argument.__name__, argument)
-                artifact = argument()
-                view = artifact.view(template.view_type)
-                view.__hide_from_garbage_collector = artifact
-                transformed_inputs[name] = view
+                if type(argument) == list:
+                    input_temp = []
+                    transformed_inputs[name] = []
+                    for arg in argument:
+                        input_temp.append(use.init_data(arg.__name__, arg))
+                        artifact = arg()
+                        view = artifact.view(template.view_type)
+                        view.__hide_from_garbage_collector = artifact
+                        transformed_inputs[name].append(view)
+                    inputs[name] = use.init_data_collection(name, list,
+                                                            *input_temp)
+                elif type(argument) == set:
+                    input_temp = []
+                    transformed_inputs[name] = set()
+                    for arg in argument:
+                        input_temp.append(use.init_data(arg.__name__, arg))
+                        artifact = arg()
+                        view = artifact.view(template.view_type)
+                        view.__hide_from_garbage_collector = artifact
+                        transformed_inputs[name].add(view)
+                    inputs[name] = use.init_data_collection(name, set,
+                                                            *input_temp)
```

Comment:
I would like to see these two branches collapsed, as the only difference appears to be in `init_data_collection` where you should be able to just pass in `type(argument)` instead of `set`/`list`